pkg/cri/sbserver/container_remove.go

```diff
@@ -33,19 +33,26 @@ import (
 // RemoveContainer removes the container.
 func (c *criService) RemoveContainer(ctx context.Context, r *runtime.RemoveContainerRequest) (_ *runtime.RemoveContainerResponse, retErr error) {
 	start := time.Now()
-	container, err := c.containerStore.Get(r.GetContainerId())
+	ctrID := r.GetContainerId()
+	container, err := c.containerStore.Get(ctrID)
 	if err != nil {
 		if !errdefs.IsNotFound(err) {
-			return nil, fmt.Errorf("an error occurred when try to find container %q: %w", r.GetContainerId(), err)
+			return nil, fmt.Errorf("an error occurred when try to find container %q: %w", ctrID, err)
 		}
 		// Do not return error if container metadata doesn't exist.
-		log.G(ctx).Tracef("RemoveContainer called for container %q that does not exist", r.GetContainerId())
+		log.G(ctx).Tracef("RemoveContainer called for container %q that does not exist", ctrID)
 		return &runtime.RemoveContainerResponse{}, nil
 	}
 	id := container.ID
 	i, err := container.Container.Info(ctx)
 	if err != nil {
-		return nil, fmt.Errorf("get container info: %w", err)
```

Comment:
for the safety, I think we should check the error here. If it is really 404, we can safe to delete the entry in memory.